### PR TITLE
Add initial version

### PR DESCRIPTION
**Pull request by mirland**
# Description
It includes an authentication flow using a [public api](https://www.appsloveworld.com/sample-rest-api-url-for-testing-with-authentication)

# Preview

https://user-images.githubusercontent.com/8472881/147110803-9ac4b182-d4d8-4885-a5f1-d2fe661172b9.MOV

### CODE REVIEW

**Review by nicolantean — APPROVED** [review]

File: lib/core/source/common/http_service.dart [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  HttpServiceDio() {
+    final options =
+        BaseOptions(baseUrl: 'http://restapi.adequateshop.com/api/');
+    _dio = Dio(options);
+    // dio.interceptors.add(DioCacheManager(
+    //   CacheConfig(
+    //     baseUrl: baseUrl,
+    //   ),
+    // ).interceptor);
+    // dio.interceptors.add(AuthInterceptor());
```
Comment (nicoache1) [inline lib/core/source/common/http_service.dart:59]: Why this commented?

**Comment by nicolantean** [comment]
> # Description
> It includes an authentication flow using a [public api](https://www.appsloveworld.com/sample-rest-api-url-for-testing-with-authentication)
> 
> # Preview
>  RPReplay_Final1640183832.MOV

Nice! You rock! 🚀

File: lib/core/source/auth_remote_source.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import 'package:flutter_template/core/source/common/http_service.dart';
+
+class AuthRemoteSource {
+  final HttpServiceDio _httpService;
+
+  AuthRemoteSource({required HttpServiceDio httpService})
+      : _httpService = httpService;
+
+  Future<SignInResponse> signIn(String email, String password) async {
+    final response = await _httpService.post('authaccount/login',
```
Comment (nicoache1) [inline lib/core/source/auth_remote_source.dart:12]: maybe static variables for the routes?

**Review by nicoache1 — APPROVED** [review]

File: lib/core/source/common/http_service.dart (hunk 1 above)
Comment (mirland) [reply lib/core/source/common/http_service.dart:59]: we should do that in the future, but I think we should remove this comment. Then we can include it

File: lib/core/source/auth_remote_source.dart (hunk 2 above)
Comment (mirland) [reply lib/core/source/auth_remote_source.dart:12]: I didn't use it because it's an example, but it makes sense!

**Pull request merged**